file/src/main/java/org/apache/zeppelin/file/HDFSFileInterpreter.java

```diff
@@ -0,0 +1,225 @@
+      return str;
+    }
+  }
+
+  /**
+   * Status of one file
+   *
+   * matches returned JSON
+   */
+  public class SingleFileStatus {
+    public OneFileStatus FileStatus;
+  }
+
+  /**
+   * Status of all files in a directory
+   *
+   * matches returned JSON
+   */
+  public class MultiFileStatus {
+    public OneFileStatus[] FileStatus;
+  }
+
+  /**
+   * Status of all files in a directory
+   *
+   * matches returned JSON
+   */
+  public class AllFileStatus {
+    public MultiFileStatus FileStatuses;
+  }
+
+  // tests whether we're able to connect to HDFS
+
+  private void testConnection() {
+    try {
+      if (isDirectory("/"))
+        logger.info("Successfully created WebHDFS connection");
+    } catch (Exception e) {
+      logger.error("testConnection: Cannot open WebHDFS connection. Bad URL: " + "/", e);
+      exceptionOnConnect = e;
+    }
+  }
+
+  @Override
+  public void open() {
+    testConnection();
+  }
+
+  @Override
+  public void close() {
+  }
+
+  private String listDir(String path) throws Exception {
+    return cmd.runCommand(cmd.listStatus, path, null);
+  }
+
+  private String listPermission(OneFileStatus fs){
+    String s = "";
+    s += fs.type.equalsIgnoreCase("Directory") ? 'd' : '-';
+    int p = Integer.parseInt(fs.permission, 16);
+    s += ((p & 0x400) == 0) ? '-' : 'r';
+    s += ((p & 0x200) == 0) ? '-' : 'w';
+    s += ((p & 0x100) == 0) ? '-' : 'x';
+    s += ((p & 0x40)  == 0) ? '-' : 'r';
+    s += ((p & 0x20)  == 0) ? '-' : 'w';
+    s += ((p & 0x10)  == 0) ? '-' : 'x';
+    s += ((p & 0x4)   == 0) ? '-' : 'r';
+    s += ((p & 0x2)   == 0) ? '-' : 'w';
+    s += ((p & 0x1)   == 0) ? '-' : 'x';
+    return s;
+  }
+  private String listDate(OneFileStatus fs) {
+    return new SimpleDateFormat("yyyy-MM-dd HH:mm").format(new Date(fs.modificationTime));
+  }
+  private String ListOne(String path, OneFileStatus fs) {
+    if (args.flags.contains(new Character('l'))) {
+      String s = "";
+      s += listPermission(fs) + "\t ";
+      s += ((fs.replication == 0) ? "-" : fs.replication) + "\t ";
+      s += fs.owner + "\t ";
+      s += fs.group + "\t ";
+      s += fs.length + "\t ";
+      s += listDate(fs) + " GMT\t ";
+      s += (path.length() == 1) ? path + fs.pathSuffix : path + '/' + fs.pathSuffix;
+      return s;
+    }
+    return fs.pathSuffix;
+  }
+
+  public String listAll(String path) {
+    String all = "";
+    if (exceptionOnConnect != null)
+      return all;
+    try {
+      String sfs = listDir(path);
+      if (sfs != null) {
+        AllFileStatus allFiles = gson.fromJson(sfs, AllFileStatus.class);
+
+        if (allFiles != null &&
+            allFiles.FileStatuses != null &&
+            allFiles.FileStatuses.FileStatus != null)
+        {
+          for (OneFileStatus fs : allFiles.FileStatuses.FileStatus)
+            all = all + ListOne(path, fs) + '\n';
+        }
+      }
+    } catch (Exception e) {
+      logger.error("listall: listDir " + path, e);
```

Comment:
should this re-throw the exception then?
because an exception here will show up in the notebook paragraph result but the logger.error() will go to a log file that user will not easily have access to